src/pages/gradient-background.ts

```diff
@@ -48,27 +47,31 @@ export function gradientBackgroundGenerator(
   type: 'newResults' | 'oldResults' | null
 ) {
   if (type === null) return;
-  // Show error when the colors are not entered.
-  var element = gradientBackgroundInputs[0];
-  var value = element.value;
-  if (value.length < 3) {
-    gradientBackgroundInputs.forEach((ele) => {
-      if (getResultBtn) {
-        getResultBtn.style.backgroundColor = 'grey';
+
+  let isColorInputFieldEmpty = false;
+
+  if (type === 'newResults') {
+    gradientBackgroundInputs.forEach((element) => {
+      if (element.value.length === 0) {
+        triggerEmptyAnimation(element);
+        (resultBtn as HTMLElement).style.backgroundColor = 'grey';
+        isColorInputFieldEmpty = true;
       }
-      getOpenSideBarButton().style.display = 'none';
-      triggerEmptyAnimation(ele);
     });
-    return;
-  } else {
-    if (getResultBtn) {
-      getResultBtn.style.backgroundColor = 'blue';
-    }
+    if (isColorInputFieldEmpty) return;
+
+    if (isColorInputFieldEmpty === false)
+      (resultBtn as HTMLElement).style.backgroundColor = 'blue';
```

Comment:
why did you change this?

Reply:
The logic was improved to check for empty input boxes if additional input boxes are added in gradient background generator. Currently if we add two more input box for colors but we select the colors for only the first two input boxes, then on clicking the `get result` button we can view a blank result page.

![image](https://github.com/Dun-sin/Code-Magic/assets/54050465/93ffd0d3-de7b-45f0-842e-b9c1bf7a2bed)

![image](https://github.com/Dun-sin/Code-Magic/assets/54050465/28350761-0e47-4e20-8386-1a05b35111af)

Reply:
not my question, i'm asking why the previous way of changing the background color was changed

Reply:
We can use your approach of changing the background color. There is nothing wrong in it. However since the resultBtn is a HTML element which already exists in the html code we can be sure that resultBtn won't be null at any point of time if not the button is removed explicitly. So we can take advantage of the typescript type assertion.

Reply:
if there's nothing wrong. don't change it, revert back please